src/main/java/com/minecolonies/coremod/blocks/BlockPaperwall.java

```diff
@@ -0,0 +1,282 @@
+            new AxisAlignedBB(0.4375D, 0.0D, 0.0D, 1.0D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.0D, 1.0D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 1.0D, 1.0D)};
+
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float BLOCK_HARDNESS = 3F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME = "blockPaperwall";
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float RESISTANCE = 1F;
+
+    BlockPaperwall()
+    {
+        super(Material.WOOD);
+        initBlock();
+    }
+
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        GameRegistry.register(this);
+        GameRegistry.register((new ItemBlock(this)).setRegistryName(this.getRegistryName()));
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+    }
+
+    public void addCollisionBoxToList(@NotNull IBlockState iBlockState,
+                                      @NotNull final World worldIn,
+                                      @NotNull final BlockPos pos,
+                                      @NotNull final AxisAlignedBB entityBox,
+                                      @NotNull final List<AxisAlignedBB> collidingBoxes,
+                                      @Nullable final Entity entityIn)
+    {
+        iBlockState = this.getActualState(iBlockState, worldIn, pos);
+        addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[0]);
+
+        if (iBlockState.getValue(NORTH))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.NORTH)]);
+        }
+
+        if (iBlockState.getValue(SOUTH))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.SOUTH)]);
+        }
+
+        if (iBlockState.getValue(EAST))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.EAST)]);
+        }
+
+        if (iBlockState.getValue(WEST))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.WEST)]);
+        }
+    }
+    private static int getBoundingBoxIndex(final EnumFacing side)
+    {
+        return 1 << side.getHorizontalIndex();
+    }
+
+    @NotNull
+    public AxisAlignedBB getBoundingBox(@NotNull IBlockState blockState1,
+                                        @NotNull final IBlockAccess source,
+                                        @NotNull final BlockPos pos)
+    {
+        blockState1 = this.getActualState(blockState1, source, pos);
```

Comment:
once again the reassign, these can cause issues. also can we get rid of the "1" in the var name please? (Not important, but while you're at it since you're already on that line) =D